### PR TITLE
tests: Address incorrect built-in function check

### PR DESCRIPTION
**Pull request by charlieegan3**
After seeing https://github.com/StyraInc/regal/actions/runs/10996729351/job/30530568473, I had another look at this one. I spotted the built-in is incorrect here, I think? Not thinking this is the main fix, but this should log to make it clear when we see the next failure.

### CODE REVIEW

File: internal/lsp/server_test.go [hunk 1]
```diff
@@ -336,30 +336,29 @@ capabilities:
...
 		select {
 		case <-timeout.C:
 			t.Fatalf("timed out waiting for builtins map to be updated")
-		default:
+		case <-ticker.C:
 			bis := rego.GetBuiltins()
 
 			// Search for a builtin we know is only in the EOPA capabilities.
-			if _, ok := bis["startswith"]; ok {
+			if _, ok := bis["neo4j.query"]; ok {
```
Comment (charlieegan3) [inline internal/lsp/server_test.go:357]: This is the main update.

File: internal/lsp/server_test.go [hunk 2]
```diff
@@ -336,30 +336,29 @@ capabilities:
...
-	// NOTE(charles): the configuration is updated asynchronously from the
-	// thread the test is running in. This check prevents a race condition
-	// where the completion test runs before rego.builtIns is updated which
-	// was causing flaky tests in CI.
+	// manually inspect the server's list of builtins to ensure that the EOPA
+	// capabilities were loaded correctly.
 	timeout = time.NewTimer(defaultTimeout)
 	defer timeout.Stop()
 
+	ticker := time.NewTicker(100 * time.Millisecond)
```
Comment (charlieegan3) [inline internal/lsp/server_test.go:344]: Using a ticker is just for consistency with the below check

**Review by anderseknert — APPROVED** [review]

**Comment by charlieegan3** [comment]
Hmm, TestGetInlayHintsAstCall... I hadn't seen that in a while... Will merge for now and see about this one tomorrow.

**Comment by anderseknert** [comment]
Haven't heard that name in years!

**Pull request merged**